src/classes/frDonation.cls

```diff
@@ -143,7 +143,7 @@ public class frDonation extends frModel {
         Boolean alreadyMappedToPledge = [SELECT COUNT() FROM Opportunity 
                                          WHERE fr_Id__c = :funraiseId AND Funraise_Pledge__c != null] > 0;
         if(!isPledge && !alreadyMappedToPledge) {
-            Pledge__c pledge = frPledge.findActive(opp.fr_Donor__c);
+            Pledge__c pledge = frPledge.findActive(opp.fr_Donor__c, opp.CloseDate);
```

Comment:
opp.CloseDate is a required field to save an opportunity, and our default mappings set this value to the donationDate of the donation.